scripts/validate-commits.sh

```
@@ -0,0 +1,298 @@
+log_error() {
+  echo "$1" >&2
+}
+
+# Check if user is a bot
+is_bot_user() {
+  local email="$1"
+  local name="$2"
+  
+  for pattern in "${BOT_PATTERNS[@]}"; do
+    # Convert email and name to lowercase for case-insensitive matching
+    local email_lower=$(echo "$email" | tr '[:upper:]' '[:lower:]')
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)